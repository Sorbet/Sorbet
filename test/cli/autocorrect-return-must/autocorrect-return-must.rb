# typed: true

extend T::Sig

sig {returns(String)}
def direct_return
  T.let("", T.nilable(String))
end

sig {params(x: T.nilable(String)).returns(String)}
def return_arg(x)
  x
end

sig {params(x: T.nilable(String)).returns(String)}
def return_uninitialized(x)
  if T.unsafe(nil)
    a = ""
  end
  a
end

sig {params(x: T.nilable(String)).returns(String)}
def implicit_return(x)
  if T.unsafe(nil)
    ""
  end
end
