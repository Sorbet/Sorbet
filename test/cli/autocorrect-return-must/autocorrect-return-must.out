# typed: true

extend T::Sig

sig {returns(String)}
def direct_return
  T.must(T.let("", T.nilable(String)))
end

sig {params(x: T.nilable(String)).returns(String)}
def return_arg(x)
  T.must(x)
end

sig {params(x: T.nilable(String)).returns(String)}
def return_uninitialized(x)
  if T.unsafe(nil)
    a = ""
  end
  T.must(a)
end

sig {params(x: T.nilable(String)).returns(String)}
T.must(def implicit_return(x)
  if T.unsafe(nil)
    ""
  end
end)
